### PR TITLE
add pass through for vendors and items

### PR DESCRIPTION
**Pull request by rpalloni**
**Are you a current Fivetran customer?** 
Yes

**What change(s) does this PR introduce?** 
It allows to extend the fields in stg models as regards vendors and items

**Did you update the CHANGELOG?** 
- [ x] Yes

**Does this PR introduce a breaking change?**
- [ ] Yes (please provide breaking change details below.)
- [x ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 
- [ x] Yes

**Is this PR in response to a previously created Bug or Feature Request**
- [x ] Yes, Issue/Feature [https://github.com/fivetran/dbt_netsuite/issues/46]
- [ ] No 

**How did you test the PR changes?** 
- [ ] CircleCi  
- [ x] Local (please provide additional testing details below)
Edit package in my local project and run the model creation. Successful after introducing fixing.

**Select which warehouse(s) were used to test the PR**
- [ ] BigQuery
- [ ] Redshift
- [ x] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# dbt_netsuite_source v0.5.2
+
+## Features
+- Added this changelog to capture iterations of the package!
+- Added the `vendors_pass_through_columns` variable to be leveraged within the `stg_netsuite__vendors` model. This variable will allow users to pass through fields which are not natively included to the package staging model. This variable is empty by default and may be overridden within a users root `dbt_project.yml`.
+- Added the `items_pass_through_columns` variable to be leveraged within the `stg_netsuite__items` model. This variable will allow users to pass through fields which are not natively included to the package staging model. This variable is empty by default and may be overridden within a users root `dbt_project.yml`.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: Can you link this PR in the CHANGELOG notes? See below for how we did it in past entries.

File: models/netsuite/stg_netsuite__items.sql [hunk 2]
```diff
@@ -36,6 +36,9 @@ final as (
         salesdescription as sales_description,
         _fivetran_deleted
 
+        --The below macro adds the fields defined within your items_pass_through_columns variable into the staging model
+        {{ fivetran_utils.fill_pass_through_columns('items_pass_through_columns') }}
```
Comment (fivetran-joemarkiewicz) [inline models/netsuite/stg_netsuite__items.sql:40]: We will also want to add a similar macro the the respective `get_items_columns` macro. See how we did it for the accounts macro:
https://github.com/fivetran/dbt_netsuite_source/blob/799a3d9efc6d2ce258e35365d0b8f102f36ae24d/macros/get_accounts_columns.sql#L15

In particular you will want to add the following in the same place:
```yml
{{ fivetran_utils.add_pass_through_columns(columns, var('items_pass_through_columns')) }} 
```

File: models/netsuite/stg_netsuite__vendors.sql [hunk 3]
```diff
@@ -36,6 +36,9 @@ final as (
         vendor_type_id,
         _fivetran_deleted
 
+        --The below macro adds the fields defined within your vendors_pass_through_columns variable into the staging model
+        {{ fivetran_utils.fill_pass_through_columns('vendors_pass_through_columns') }}
```
Comment (fivetran-joemarkiewicz) [inline models/netsuite/stg_netsuite__vendors.sql:40]: Same comment as the other:

We will also want to add a similar macro the the respective `get_vendors_columns` macro. See how we did it for the accounts macro:
https://github.com/fivetran/dbt_netsuite_source/blob/799a3d9efc6d2ce258e35365d0b8f102f36ae24d/macros/get_accounts_columns.sql#L15

In particular you will want to add the following in the same place:
```yml
{{ fivetran_utils.add_pass_through_columns(columns, var('vendors_pass_through_columns')) }} 
```

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@rpalloni thanks again for opening this PR! I have just a few small requests to edit your PR before this will be good to go. Let me know if you have any questions!

File: CHANGELOG.md (hunk 1 above)
Comment (ropalloni) [reply CHANGELOG.md:6]: Ooops! Yes, sure #23

File: models/netsuite/stg_netsuite__items.sql (hunk 2 above)
Comment (ropalloni) [reply models/netsuite/stg_netsuite__items.sql:40]: Done

File: models/netsuite/stg_netsuite__vendors.sql (hunk 3 above)
Comment (ropalloni) [reply models/netsuite/stg_netsuite__vendors.sql:40]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (rpalloni) [reply CHANGELOG.md]: Ooops! Yes sure #23

File: models/netsuite/stg_netsuite__items.sql (hunk 2 above)
Comment (rpalloni) [reply models/netsuite/stg_netsuite__items.sql:40]: Done

File: models/netsuite/stg_netsuite__vendors.sql (hunk 3 above)
Comment (rpalloni) [reply models/netsuite/stg_netsuite__vendors.sql:40]: Done

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@rpalloni thanks so much for updating your PR and for contributing to the package. These variables will be a great addition for folks to be able to leverage! 🏅 

I have approved the PR, but will be merging it into our existing `MagicBot/dbt-utils-cross-db-migration` branch as we are planning a large scale migration of all our packages for dbt utils v1.0.0 and dbt-core v1.3.0 compatibility. As such, I am folding your changes into this effort.

If you need to use your changes immediately, you can leverage your forked branch and wait to swap over to the official hub version once we roll these changes out. Let me know if you have any questions. Thanks again!!

**Comment by rpalloni** [comment]
Ok cool! Found a temporary fix in the meanwhile so I think I'll wait for the large scale migration. Any idea on when it will be?
Thanks anyway

**Comment by fivetran-joemarkiewicz** [comment]
Awesome! Next week our team will be at the Coalesce Conference and we will plan to make a team wide effort to roll these out the following week. However, this large scale release is all dependent on when [dbt-utils v1.0.0](https://github.com/dbt-labs/dbt-utils/releases) officially releases.

**Comment by ropalloni** [comment]
Hi @fivetran-joemarkiewicz , any news on this?
Seems like [dbt-utils v1.0.0](https://github.com/dbt-labs/dbt-utils/releases) is up since two weeks but old version is still there 
image

Thanks

**Comment by fivetran-joemarkiewicz** [comment]
Hey @ropalloni apologies for the late reply as I am just getting back from the holiday PTO.

Thanks for asking about this, and we will be releasing these updates this week! You should see all our packages updates before the end of the week. Thanks!

**Comment by ropalloni** [comment]
Ok now it works. However you did not align **netsuite2** with the same fix....

**Comment by fivetran-joemarkiewicz** [comment]
Hi @ropalloni thanks for raising this to me. You are correct that we missed in this PR to persist the items and vendors passthrough columns to the netsuite2 models. 

We can account for this in the next release of the netsuite package. Is this currently a blocker for you?

**Comment by ropalloni** [comment]
When will be next release?

**Comment by ropalloni** [comment]
This week or next week would be perfect! Btw it's just adding the same macro in those models.

**Comment by fivetran-joemarkiewicz** [comment]
Sounds great! Yeah this should be a pretty simple update, but want to be realistic with the other changes (across our other packages) that are taking place in our current sprint. I just created a new issue (above) for us to be able to track this bug fix.

I also marked it as a good first issue in case others come across and want to contribute to the package before my team is able to get to it.

**Comment by ropalloni** [comment]
Hi @fivetran-joemarkiewicz ! Do you have any updates on this? Thanks

**Comment by fivetran-joemarkiewicz** [comment]
Hey @ropalloni the issue I linked above to include the passthrough columns is currently in progress. I would reference Issue #25 and the corresponding PR #27 for the progress on this addition.

**Pull request merged**